service/src/main/java/org/cbioportal/service/impl/GenesetDataServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2016 The Hyve B.V.
+ * This code is licensed under the GNU Affero General Public License (AGPL),
+ * version 3, or (at your option) any later version.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.service.impl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.cbioportal.model.GenesetGeneticAlteration;
+import org.cbioportal.model.GenesetGeneticData;
+import org.cbioportal.model.GeneticProfile;
+import org.cbioportal.model.Sample;
+import org.cbioportal.persistence.GeneticDataRepository;
+import org.cbioportal.persistence.SampleListRepository;
+import org.cbioportal.service.GenesetDataService;
+import org.cbioportal.service.GeneticProfileService;
+import org.cbioportal.service.SampleService;
+import org.cbioportal.service.exception.GeneticProfileNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class GenesetDataServiceImpl implements GenesetDataService {
+
+    @Autowired
+    private GeneticDataRepository geneticDataRepository;
```

Comment:
Looking at it again, I think this could lead to some undesired features. Here are some thoughts:
- `geneticDataRepository.getGenesetGeneticAlterations` and `geneticDataRepository.getCommaSeparatedSampleIdsOfGeneticProfile` return CSV string with the values, something that is a bit ugly about our data model and that we would like to keep hidden from outer layers. Exposing these in the service layer would defeat this goal. 
- When I first wrote this code, I intended `GeneticDataRepository` to be a "shared" repository layer for all genetic data. That is why I added `getGenesetGeneticAlterations` there as well. As such, it should be fine to use it here in `GenesetDataServiceImpl` as well as in `GeneticDataServiceImpl` (maybe this last one should be renamed to `GeneGeneticDataServiceImpl` to keep it more consistent). Alternatively, we could introduce `GenesetDataRepository` instead of having both gene and geneset services share the same `GeneticDataRepository`.